### PR TITLE
nydus-image: fix inspect to get correct path of rafs v6 file

### PR DESCRIPTION
**Pull request by sctb512**
Because rafs v6 doesn't support get_parent, the prefetch and icheck command of inspect will cause error. We fixed it by handling get_file_name get_file_name and path_from_ino for rafs v6 files separately. 

For hardlink, we will get all the paths of file:
image

This PR is addressing #768 

Signed-off-by: Bin Tang <tangbin.bin@bytedance.com>

### CODE REVIEW

File: rafs/src/metadata/direct_v6.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ use std::slice;
 use std::sync::Arc;
 
 use arc_swap::{ArcSwap, Guard};
+use nydus_utils::metrics::Inode;
```
Comment (jiangliu) [inline rafs/src/metadata/direct_v6.rs:36]: This change seems wrong, we should use Inode from crate::metadata::Inode

File: rafs/src/metadata/direct_v6.rs [hunk 2]
```diff
@@ -1159,14 +1160,49 @@ impl RafsInode for OndiskInodeWrapper {
         match self.parent_inode.get() {
             Some(parent) => parent,
             None => {
-                debug_assert!(self.is_dir());
```
Comment (jiangliu) [inline rafs/src/metadata/direct_v6.rs:1162]: It would be better to keep `parent()` as is, and modify `nydus-image` to call `file_parent_inode()` directly.

**Comment by bergwolf** [comment]
rafs v5 can have i_parent for every inode because it duplicates hardlink inodes and each would have different i_parent. For rafs v6, this is unsupported by design. If we still want to have such functionality, we need to be aware that for rafs v6 hardlinks, it is unreliable to find inode's parent by traversing from the root inode, and specially deal with it in `nydus-image inspect`.

File: rafs/src/metadata/direct_v6.rs (hunk 2 above)
Comment (sctb512) [reply rafs/src/metadata/direct_v6.rs:1162]: Ok, i got it.

**Comment by sctb512** [comment]
> nydus-image inspect

Ok, maybe deal with it in `nydus-image inspect` is better.

**Comment by jiangliu** [comment]
> > nydus-image inspect
> 
> Ok, maybe deal with it in `nydus-image inspect` is better.

The fundamental issue is that v6 doesn't support "get_parent", so `nydus-image inspect` could only do best effort to find a parent. And the runtime can't guarenttee `get_parent()`

**Comment by sctb512** [comment]
> > > nydus-image inspect
> > 
> > 
> > Ok, maybe deal with it in `nydus-image inspect` is better.
> 
> The fundamental issue is that v6 doesn't support "get_parent", so `nydus-image inspect` could only do best effort to find a parent. And the runtime can't guarenttee `get_parent()`

I got it.
It means try to find parent only in `nydus-image inspect`, right? 
For hardlink, if needed, we can traverse it completely to find all parent?

File: rafs/src/metadata/direct_v6.rs (hunk 1 above)
Comment (sctb512) [reply rafs/src/metadata/direct_v6.rs:36]: Sorry, I fixed it already.

**Pull request merged**